### PR TITLE
Alertmanager ruler proxy

### PR DESCRIPTION
**Pull request by dogukanteber**
* Alertmanager and ruler proxies are implemented

### CODE REVIEW

File: gateway/gateway.go [hunk 1]
```diff
@@ -38,6 +40,23 @@ var defaultQueryFrontendAPIs = []string{
 	"/api/prom/api/v1/status/buildinfo",
 }
 
+var defaultAlertManagerAPIs = []string{
+	"/alertmanager",
```
Comment (friedrichg) [inline gateway/gateway.go:44]:
```suggestion
	"/alertmanager/",
```
Need everything that is inside /alertmanager/

File: gateway/gateway.go [hunk 2]
```diff
@@ -38,6 +40,23 @@ var defaultQueryFrontendAPIs = []string{
...
+	"/api/prom",
```
Comment (friedrichg) [inline gateway/gateway.go:45]:
```suggestion
	"/api/prom/alertmanager/",
```
But needs testing. I think the docs are wrong https://cortexmetrics.io/docs/api/#alertmanager-ui

**Review by friedrichg — CHANGES_REQUESTED** [review]

File: gateway/gateway.go (hunk 2 above)
Comment (dogukanteber) [reply gateway/gateway.go:45]: How can I test this endpoint?

Comment (friedrichg) [reply gateway/gateway.go:45]: I would start cortex with target=alertmanager and see what paths are available in cortex

**Comment by friedrichg** [comment]
this needs a rebase

File: gateway/gateway.go (hunk 2 above)
Comment (dogukanteber) [reply gateway/gateway.go:45]: You are right, the documentation is wrong. I got 404 when I requested `/api/prom` endpoint. However, `/api/prom/alertmanager` does not work either. I got 404 when I requested that endpoint, too. 

Everything that is related to alertmanager ui is under `/alertmanager/`. I propose to remove `/api/prom` altogether and only implement `/alertmanager/`. If we come to an agreement, I will open a PR to fix the documentation as well

Comment (friedrichg) [reply gateway/gateway.go:45]: just finish this PR then with /alertmanager only.

**Comment by dogukanteber** [comment]
This branch has conflicts that are too hard to fix. I opened another PR for alertmanager and will open another one for ruler. Because of that, closing this one.

**Pull request closed without merging**